### ISSUE TITLE
UnusedInputError in Tutorial

### ISSUE THREAD

**Issue by rasmi** [Theano/Theano#4980]
UPDATE: This only affects the 0.8.x and not current dev or 9.x versions. The error was fixed in 0.8.2 and the documentation will be updated accordingly.

Following the tutorial section on [Copying Functions](http://deeplearning.net/software/theano/tutorial/examples.html#copying-functions), the example shown raises an error:
```
---------------------------------------------------------------------------
UnusedInputError                          Traceback (most recent call last)
<ipython-input-59-5d1acb597345> in <module>()
----> 1 null_accumulator = accumulator.copy(delete_updates=True)

/Library/Python/2.7/site-packages/theano/compile/function_module.pyc in copy(self, share_memory, swap, delete_updates, name, profile)
    719                                 # can contain inplace. DebugMode check
    720                                 # that.
--> 721                                 accept_inplace=True,
    722                                 ).create(input_storage,
    723                                          storage_map=new_storage_map)

/Library/Python/2.7/site-packages/theano/compile/function_module.pyc in __init__(self, inputs, outputs, mode, accept_inplace, function_builder, profile, on_unused_input, fgraph, output_keys)
   1413 
   1414         # Check if some input variables are unused
-> 1415         self._check_unused_inputs(inputs, outputs, on_unused_input)
   1416 
   1417         # Make a list of (SymbolicInput|SymblicInputKits, indices,

/Library/Python/2.7/site-packages/theano/compile/function_module.pyc in _check_unused_inputs(self, inputs, outputs, on_unused_input)
   1551                 elif on_unused_input == 'raise':
   1552                     raise UnusedInputError(msg % (inputs.index(i),
-> 1553                                                   i.variable, err_msg))
   1554                 else:
   1555                     raise ValueError("Invalid value for keyword "

UnusedInputError: theano.function was asked to create a function computing outputs given certain inputs, but the provided input variable at index 0 is not part of the computational graph needed to compute the outputs: inc.
To make this error into a warning, you can pass the parameter on_unused_input='warn' to theano.function. To disable it completely, use on_unused_input='ignore'.
```

The `accumulator` function must have the `on_unused_input='ignore'` option added, since the `null_accumulator` function does not use the `inc` argument.

**Comment by Sentient07** [comment]
No, it doesn't happen with my Theano. Did you happen to miss the update parameter ?

**Comment by Sentient07** [comment]
Oh! I now see what might have happened. 
```
null_accumulator = accumulator.copy(delete_updates=True)
```
when you try to copy with `delete_updates=True`, your function will not have updates, rendering the input variable to be useless. Change that and the code will work

**Comment by rasmi** [comment]
This was fixed after the 0.8.2 release. Documentation has been updated [here](https://github.com/Theano/Theano/pull/4988).
